chain/client/tests/process_blocks.rs

```diff
@@ -825,3 +828,29 @@ fn test_minimum_gas_price() {
     let block = env.clients[0].chain.get_block_by_height(100).unwrap();
     assert!(block.header.inner_rest.gas_price >= min_gas_price);
 }
+
+#[test]
+fn test_garbage_collection() {
+    let store = create_test_store();
+    let epoch_length = 2;
+    let mut genesis = Genesis::test(vec!["test0", "test1"], 1);
+    genesis.config.epoch_length = epoch_length;
+    let runtimes: Vec<Arc<dyn RuntimeAdapter>> = vec![Arc::new(near::NightshadeRuntime::new(
+        Path::new("."),
+        store,
+        Arc::new(genesis),
+        vec![],
+        vec![],
+    ))];
+    let mut chain_genesis = ChainGenesis::test();
+    chain_genesis.epoch_length = epoch_length;
```

Comment:
I'm not sure how is it possible, but can we imitate not finalizing epoch in TestEnv easily? If yes, please add one more parameter to the test `is_finalizing` to make sure we're not GCing in Nightshade non-finalized epochs.

Reply:
it's inconvenient to combine the two so I made a separate test for this